### PR TITLE
NEO-1804 PR #2

### PR DESCRIPTION
**Pull request by mrazian85**
[Link to updated components in Deploy Preview](https://deploy-preview-242--neo-react-library-storybook.netlify.app/?path=/docs/components-notification--docs)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

This is PR #2 for [NEO-1804](https://jira.forge.avaya.com/browse/NEO-1804), the Notification Component re-work. This PR includes the following changes:

- Added logic to conditionally render truncation toggle based on width of Notification and re-size text appropriately
- Added logic to allow for rendering the Notification at different locations on the screen based on whether it is inline vs not
- Further CSS styling tweaks
- Added logic to pass in default as well as localized values for all `aria-label` attributes in Component where necessary
- Shored up testing file to account for new Component logic

**EXAMPLE IMAGE(S):**
Screen Shot 2023-08-22 at 3 30 56 PM

**EXAMPLE FUNCTIONALITY GIF:**
Truncation toggle behaviour

### CODE REVIEW

File: src/components/Notification/Notification.tsx [hunk 1]
```diff
@@ -186,6 +226,10 @@ export const Notification = ({
       </div>
     </div>
   );
+
+  return isInline
+    ? notification
+    : ReactDOM.createPortal(notification, document.body);
```
Comment (yangAtSpoken) [inline src/components/Notification/Notification.tsx:232]: We should still use PopupManager to position the notification when it is not inline.  In storybook, we have a couple of stories under PopupManager -> Notification. There is also a hook, usePopup, that simplifies this a bit.

**Review by yangAtSpoken — CHANGES_REQUESTED** [review]
Left a comment.

File: src/components/Notification/Actions/ButtonAction/SecondaryButton.tsx [hunk 2]
```diff
@@ -1,15 +1,18 @@
 import { Button } from "components/Button";
+
 import { SecondaryButtonProps } from "./ButtonActionTypes";
 
 export const SecondaryButton = ({
   children,
+  notificationType,
```
Comment (joe-s-avaya) [inline src/components/Notification/Actions/ButtonAction/SecondaryButton.tsx:7]: should this have a default?

File: src/components/Notification/Notification.tsx [hunk 3]
```diff
@@ -98,14 +107,45 @@ export const Notification = ({
...
+  const [needsTruncation, setNeedsTruncation] = useState(false);
+
+  const notificationTranslations = useMemo(() => {
+    return {
+      ...defaultTranslations,
+      ...translations,
+    };
+  }, [translations]);
+
+  function handleResize() {
```
Comment (joe-s-avaya) [inline src/components/Notification/Notification.tsx:120]: nit: for consistency I recommend using arrow methods
```suggestion
  const handleResize = () => {
```

File: src/components/Notification/Notification.tsx [hunk 4]
```diff
@@ -77,11 +80,17 @@ export const Notification = ({
...
   ...rest
 }: NotificationProps) => {
   const icon = "icon" in rest ? rest.icon : null;
   const [closed, setClosed] = useState(false);
-  const internalActions = createActions(actions, type, setClosed);
+  const internalActions = createActions(
+    actions,
+    type,
+    setClosed,
+    translations?.closeAction,
```
Comment (joe-s-avaya) [inline src/components/Notification/Notification.tsx:92]: your translations all have defaults yes? you shouldn't need the `?`

File: src/components/Notification/ToggleNotification.stories.tsx [hunk 5]
```diff
@@ -28,6 +28,7 @@ export const AlertShow = () => {
         <Notification
           type="alert"
           header="Alert"
+          isInline={false}
```
Comment (joe-s-avaya) [inline src/components/Notification/ToggleNotification.stories.tsx:31]: nit: maybe have `false` be the default? would that simplify the usage?

**Review by joe-s-avaya — APPROVED** [review]
There are some nits and one bug, but I won't block. They're all quick and easy so I assume that you'll take care of it before merging.

**Comment by matthewAvaya** [comment]
Two late updates from the team.
- Make both the header and the body copy to be 2 lines worth before truncation. Applies to both general and inline.
- Increase the padding between the header and the body copy to be 4 px. Applies to both general and inline.

Additional comments: 
- I'm not seeing a tooltip when hovering over the badge.

File: src/components/Notification/Notification.tsx (hunk 1 above)
Comment (mrazian85) [reply src/components/Notification/Notification.tsx]: Great point; I have restored use of PopupManager to display the notification by moving the render logic outside of the Component itself and updating the stories in `PopupManager.notification.stories.tsx`.

I have also added a default width for the Notification when used with PopUpManager and allowed the user to pass in their own width for custom styling in order to preserve the way text truncation logic is handled

File: src/components/Notification/Actions/ButtonAction/SecondaryButton.tsx (hunk 2 above)
Comment (mrazian85) [reply src/components/Notification/Actions/ButtonAction/SecondaryButton.tsx]: Good call; I have added one ✅

File: src/components/Notification/Notification.tsx (hunk 4 above)
Comment (mrazian85) [reply src/components/Notification/Notification.tsx:93]: The translations do have defaults, but these are handled by the `defaultTranslations` import

The `translations` prop itself is optional

File: src/components/Notification/Notification.tsx (hunk 3 above)
Comment (mrazian85) [reply src/components/Notification/Notification.tsx]: Actually ended up moving this method out of the Component as I had to refactor the logic to resolve some issues when using the PopUpManager for non-inline Notifications

As an aside, I think that makes this and other methods in the file easier to unit tests, although I haven't added any tests for them in this PR as it is already quite large. I think I will either create a third PR for this, or come back and add these tests after completing the work in the CSS library, as this ticket has already ballooned quite a bit due to changing AC

File: src/components/Notification/ToggleNotification.stories.tsx (hunk 5 above)
Comment (mrazian85) [reply src/components/Notification/ToggleNotification.stories.tsx:31]: Yes, good call ✅

**Comment by mrazian85** [comment]
> Two late updates from the team.
> 
> * Make both the header and the body copy to be 2 lines worth before truncation. Applies to both general and inline.
> * Increase the padding between the header and the body copy to be 4 px. Applies to both general and inline.
> 
> Additional comments:
> 
> * I'm not seeing a tooltip when hovering over the badge.

@matthewAvaya all feedback should be picked up now, except for the padding between header and body was already at 4px as far as I could tell

**Review by matthewAvaya — APPROVED** [review]
Looks great!

**Review by yangAtSpoken — APPROVED** [review]

**Pull request merged**